init.lua

```diff
@@ -305,46 +272,59 @@ local function get_recipe_fs(iX, iY, xoffset, recipe_num, recipes, show_usage)
 		0.7,
 		"craftguide_arrow.png")
 
-	if output == "BURN" then
+	if recipe.type == "fuel" then
 		fs[#fs + 1] = fmt("image[%f,%f;%f,%f;%s]",
 			output_X,
 			iY + (sfinv_only and 2.68 or 2.18),
 			1.1,
 			1.1,
 			"craftguide_fire.png")
 	else
+		local output_name = recipe.output:match("%S+")
+		local burntime = fuel_cache[output_name]
+
 		fs[#fs + 1] = fmt("item_image_button[%f,%f;%f,%f;%s;%s;]",
 			output_X,
 			iY + (sfinv_only and 2.7 or 2.2),
 			1.1,
 			1.1,
-			output,
-			output_s)
+			recipe.output,
+			output_name)
+
+		if burntime then
+			fs[#fs + 1] = get_tooltip(output_name, nil, nil, burntime)
+
+			fs[#fs + 1] = fmt("image[%f,%f;%f,%f;%s]",
+				output_X + 1,
+				iY + (sfinv_only and 2.83 or 2.33),
+				0.6,
+				0.4,
+				"craftguide_arrow.png")
+
+			fs[#fs + 1] = fmt("image[%f,%f;%f,%f;%s]",
+				output_X + 1.6,
+				iY + (sfinv_only and 2.68 or 2.18),
+				0.6,
+				0.6,
+				"craftguide_fire.png")
+		end
 	end
 
-	fs[#fs + 1] = get_tooltip(output_s)
-
-	local output_is_fuel = get_fueltime(output) > 0
-	if output_is_fuel then
-		fs[#fs + 1] = fmt("image[%f,%f;%f,%f;%s]",
-			output_X + 1,
-			iY + (sfinv_only and 2.83 or 2.33),
-			0.6,
-			0.4,
-			"craftguide_arrow.png")
-
-		fs[#fs + 1] = fmt("image[%f,%f;%f,%f;%s]",
-			output_X + 1.6,
-			iY + (sfinv_only and 2.68 or 2.18),
-			0.6,
-			0.6,
-			"craftguide_fire.png")
-	end
+	fs[#fs + 1] = fmt("button[%f,%f;%f,%f;%s;%s %u %s %u]",
+		data.iX - (sfinv_only and 2.2 or 2.6),
+		iY + (sfinv_only and 3.9 or 3.3),
+		2.2,
+		1,
+		"alternate",
+		data.show_usages and S("Usage") or S("Recipe"),
+		data.rnum,
+		S("of"),
+		#data.recipes)
 
 	return concat(fs)
 end
 
-local function make_formspec(player_name)
+local function get_formspec(player_name)
```

Comment:
Keep `make_formspec`

Reply:
I thought `get_formspec` would make more sense now that it always returns the formspec. Also there's already `sfinv.make_formspec` and it's consistent with `get_recipe_fs`.

Reply:
Keep `make_formspec` anyway. `get` feels like "showing", we are only "building" a formspec here.